src/prezzemolo/graph.py

```diff
@@ -87,6 +110,42 @@ def breadth_first_search(self, start: Vertex[ValueType], end: Vertex[ValueType])
         # start and end are type checked inside _breadth_first_search()
         return self._breadth_first_search(start, end, vertex_2_parent=None)
 
+    def _dijkstra_search(
+        self, start: Vertex[ValueType], end: Optional[Vertex[ValueType]], vertex_2_parent: Optional[Dict[Vertex[ValueType], Optional[Vertex[ValueType]]]] = None
+    ) -> bool:
+        distance: Dict[Vertex[ValueType], float] = {v: float("inf") for v in self.__vertexes}
+        distance[start] = 0.0
+        remaining_vertexes: PriorityQueue[_ShortestDistance[ValueType]] = PriorityQueue()
+        remaining_vertexes.put(_ShortestDistance(distance=0.0, vertex=start))
+        visited: Set[Vertex[ValueType]] = set()
```

Comment:
I think we don't need `visited`: we could just initialize `remaining_vertexes` with all the vertexes and their distance before the loop, and then we iterate normally removing the nearest element (as you are doing).

Reply:
Distance from what? The first vertex? How would the other distances be added if we are not putting them during the loop?

Reply:
This was my mistake: ignore this.